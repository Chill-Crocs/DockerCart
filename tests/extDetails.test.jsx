/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ExtDetails from '../client/components/ExtDetails';

const extDetails = {
  sales: 400,
  availability: true,
  description: 'A description',
};

describe('extDetails render', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ExtDetails extDetails={extDetails} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('onClick', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ExtDetails extDetails={extDetails} />);
  });
  test('should change state when details collap is clicked', () => {
    const event = {
      target: {
        className: 'detailsCollapsible',
      },
    };
    const instance = wrapper.instance();
    jest.spyOn(instance, 'collapseOnClick');
    instance.collapseOnClick(event);
    expect(wrapper.state('detailsClicked')).toEqual(false);
  });
  test('should change state when description collap is clicked', () => {
    const event = {
      target: {
        className: 'descriptionCollapsible',
      },
    };
    const instance = wrapper.instance();
    jest.spyOn(instance, 'collapseOnClick');
    instance.collapseOnClick(event);
    expect(wrapper.state('descriptionClicked')).toEqual(false);
  });
  test('should change state when shipping collap is clicked', () => {
    const event = {
      target: {
        className: 'shippingCollapsible',
      },
    };
    const instance = wrapper.instance();
    jest.spyOn(instance, 'collapseOnClick');
    instance.collapseOnClick(event);
    expect(wrapper.state('shippingClicked')).toEqual(false);
  });
  test('should change state when description button is clicked for content hide', () => {
    const event = {
      target: {
        className: 'descriptionButton',
      },
    };
    const instance = wrapper.instance();
    jest.spyOn(instance, 'collapseOnClick');
    instance.collapseOnClick(event);
    expect(wrapper.state('descriptionExpand')).toEqual('contentExpand');
  });
  test('should change state when description button is clicked for content expand', () => {
    const event = {
      target: {
        className: 'descriptionButton',
      },
    };
    const instance = wrapper.instance();
    jest.spyOn(instance, 'collapseOnClick');
    instance.collapseOnClick(event);
    instance.collapseOnClick(event);
    expect(wrapper.state('shader')).toEqual('shaderOn');
  });
  test('should change state on random classname', () => {
    const event = {
      target: {
        className: 'aButton',
      },
    };
    const instance = wrapper.instance();
    instance.collapseOnClick(event);
    expect(wrapper.state('detailsClicked')).toBe(true);
    expect(wrapper.state('descriptionClicked')).toBe(true);
    expect(wrapper.state('shippingClicked')).toBe(true);
    expect(wrapper.state('descriptionExpand')).toEqual('contentHide');
    expect(wrapper.state('shader')).toEqual('shaderOn');
    expect(wrapper.state('expandButtonContent')).toEqual('Learn more about this item');
  });
});

const extDetails2 = {
  sales: 100,
  availability: true,
  description: 'A description',
};

const wrapper2 = mount(<ExtDetails extDetails={extDetails2} />);

describe('Low sales', () => {
  test('should not render an icon on low sales', () => {
    expect(wrapper2.find('.inline-svg-1').exists()).toBeFalsy();
    expect(wrapper2.find('.inline-svg-2').exists()).toBeFalsy();
  });
});
