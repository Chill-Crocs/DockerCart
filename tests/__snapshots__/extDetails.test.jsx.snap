// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extDetails render renders correctly 1`] = `
<div
  className="extDetails"
>
  <div
    className="extDetails-buttons"
  >
    <div
      className="buyButton"
    >
      <button
        type="button"
      >
         Buy it now 
      </button>
    </div>
    <div
      className="cartButton"
    >
      <button
        type="button"
      >
         Add to cart 
      </button>
    </div>
  </div>
  <div>
    <span
      className="inline-svg-2"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="-6 11 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-animator-group="true"
          data-animator-type="2"
        >
          <polygon
            id="cart-star1"
            points="43 8 41.9 6.1 40 5 41.9 3.9 43 2 44.1 3.9 46 5 44.1 6.1 "
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="2"
        >
          <polygon
            id="cart-star2"
            points="6 27 4.9 25.1 3 24 4.9 22.9 6 21 7.1 22.9 9 24 7.1 25.1 "
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="2"
        >
          <polygon
            id="cart-star3"
            points="43 41 41.9 39.1 40 38 41.9 36.9 43 35 44.1 36.9 46 38 44.1 39.1 "
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <g
            data-animator-group="true"
            data-animator-type="1"
          >
            <g
              data-animator-group="true"
              data-animator-type="2"
            >
              <polygon
                id="cart-basket"
                points="15.8 27 38.3 25 40 12.8 12.2 12.8 "
              />
            </g>
          </g>
        </g>
        <polygon
          id="cart-fill"
          points="15.7 18.4 20.5 18.2 20.5 14 14.6 14 "
        />
        <polygon
          id="cart-fill"
          points="23.5 25.3 28.5 24.9 28.5 20.9 23.5 21.1 "
        />
        <polygon
          id="cart-fill"
          points="23.5 18.1 28.5 17.9 28.5 14 23.5 14 "
        />
        <polygon
          id="cart-fill"
          points="17.5 25.9 20.5 25.6 20.5 21.2 16.4 21.4 "
        />
        <polygon
          id="cart-fill"
          points="31.5 17.8 37.2 17.5 37.7 14 31.5 14 "
        />
        <polygon
          id="cart-fill"
          points="31.5 24.6 36.2 24.1 36.8 20.5 31.5 20.8 "
        />
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <circle
            cx="35"
            cy="41"
            r="3"
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <circle
            cx="15"
            cy="41"
            r="3"
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <g
            data-animator-group="true"
            data-animator-type="1"
          >
            <path
              d="M40 10H13.6l-0.6-2.5L11 6H4v4h5.4l0.6 2.5L13.9 28l-1.4 5.5 1.9 2.5H38v-4H17.1l0.5-2.1L38.2 28l1.8-1.7 2-14L40 10zM20.5 25.6l-3 0.3 -1.1-4.4 4.1-0.2V25.6zM20.5 18.2l-4.8 0.2L14.6 14h5.9V18.2zM28.5 24.9l-5 0.5v-4.2l5-0.2V24.9zM28.5 17.9l-5 0.2V14h5V17.9zM36.2 24.1l-4.7 0.4v-3.8l5.3-0.2L36.2 24.1zM37.2 17.5l-5.7 0.2V14h6.2L37.2 17.5z"
            />
          </g>
        </g>
      </svg>
      <span
        className="descriptor"
      >
        <b>
          Other people want this.
          <span>
             
          </span>
        </b>
        Over 20 people have this in their carts right now.
      </span>
    </span>
  </div>
  <div
    className="details"
  >
    <span
      className="inline-svg-3"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        viewBox="-6 11 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <circle
            cx="13"
            cy="38"
            r="4"
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <circle
            cx="35"
            cy="38"
            r="4"
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <circle
            cx="13"
            cy="38"
            id="wheel-left"
            r="4"
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <circle
            cx="35"
            cy="38"
            id="wheel-right"
            r="4"
          />
        </g>
        <g
          data-animator-group="true"
          data-animator-type="0"
        >
          <g
            data-animator-group="true"
            data-animator-type="1"
          >
            <path
              d="M38.6 15c-0.6-0.3-1.1-0.6-1.7-1H28v14H4v7c0.4 0.4 0.6 0.6 1 1h2.4c0.8-2.3 3-4 5.6-4s4.8 1.7 5.6 4h10.7c0.8-2.3 3-4 5.6-4s4.8 1.7 5.6 4H43c0.4-0.4 0.6-0.6 1-1V24L38.6 15zM32 24v-6h4.9l3.6 6H32z"
            />
            <path
              d="M28 10c-0.8-0.8-1.2-1.2-2-2H6c-0.8 0.8-1.2 1.2-2 2v18h24V10z"
              id="truck-container"
            />
          </g>
        </g>
      </svg>
      <span
        className="descriptor"
      >
        <b>
          Nice choice! 
        </b>
        Enjoy free shipping to the US when you spend $35+ at this shop.
      </span>
    </span>
    <button
      className="detailsCollapsible"
      onClick={[Function]}
      type="button"
    >
      <span
        className="details-button-content"
      >
        Details
      </span>
      <span
        className="button-arrow-up"
      />
    </button>
    <div
      className="detailsBox"
    >
      <svg
        aria-hidden="true"
        fill="none"
        focusable="false"
        viewBox="-2 1 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.313 10.252l1.86-3.45-1.328-1.32-3.307-3.307a.57.57 0 00-.788 0 .57.57 0 000 .75l3.098 3.57-.533.532-4.425-4.32a.577.577 0 00-.817 0 .577.577 0 000 .795l4.14 4.628-.525.532L3.825 4.83a.555.555 0 00-.825 0 .57.57 0 000 .75l3.6 4.17-.525.532L3 7.68a.578.578 0 00-.75 0 .578.578 0 000 .817l2.775 2.768L9.27 15.51a1.498 1.498 0 002.123 0l3.75-3.713.405-.405a1.5 1.5 0 00.36-1.537L13.5 3.885l-.45.15a.578.578 0 00-.382.487l.3 3.075-2.123 3.188-.532-.533z"
          fill="#222"
        />
      </svg>
      <span
        className="details-content"
        style={
          Object {
            "marginLeft": "20px",
          }
        }
      >
        Handmade Item
      </span>
    </div>
    <button
      className="descriptionCollapsible"
      onClick={[Function]}
      type="button"
    >
      <span
        className="description-button-content"
      >
        Description
      </span>
      <span
        className="button-arrow-up"
      />
    </button>
    <div>
      <div
        className="descriptionContainer"
      >
        <div
          className="shaderOn"
        >
          <div
            className="contentHide"
          >
            A description
            <br />
            <br />
            A description
            <br />
            <br />
            A description
          </div>
        </div>
      </div>
      <div>
        <button
          className="descriptionButton"
          onClick={[Function]}
          type="button"
        >
          Learn more about this item
        </button>
      </div>
    </div>
    <button
      className="shippingCollapsible"
      onClick={[Function]}
      type="button"
    >
      <span
        className="shipping-button-content"
      >
        Shipping and return policies
      </span>
      <span
        className="button-arrow-up"
      />
    </button>
    <div
      id="shipping-collapse"
    >
      <div>
        <div
          className="dotted-shipping arrival"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          Estimated arrival
        </div>
        <span
          className="hide"
        >
          <span
            className="tooltip-container"
          >
            <span
              className="tooltip-content"
            >
              This is an estimate based on the purchase date, the seller's location,
              <span>
                 
              </span>
              and processing time, and the shipping destination and carrier. Other factors—such as shipping carrier delays or placing an order on
              <span>
                 
              </span>
              weekend/holiday—may push the arrival of your item beyond this date.
            </span>
          </span>
        </span>
        <div
          id="shippedDate"
        >
          Feb 5 - 11
        </div>
      </div>
      <div
        id="shipped-grid"
      >
        <div
          className="shipped-grid-0-0 grid-3"
          id="shipped-grid-0-0"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="grid-3"
            id="grid-0-0-content"
          >
            <span
              className="etsy-icon grid-3"
            >
              <svg
                aria-hidden="true"
                className="grid-3"
                fill="none"
                focusable="false"
                viewBox="-4 10 1.1 1.1"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.3126 10.2524L12.1726 6.80245L10.8451 5.48245L7.53761 2.17495C7.43156 2.07363 7.29053 2.01709 7.14386 2.01709C6.99719 2.01709 6.85616 2.07363 6.75011 2.17495C6.65938 2.27881 6.60938 2.41204 6.60938 2.54995C6.60938 2.68786 6.65938 2.82109 6.75011 2.92495L9.84761 6.49495L9.31511 7.02745L4.89011 2.70745C4.83647 2.6537 4.77276 2.61106 4.70262 2.58197C4.63248 2.55288 4.55729 2.53791 4.48136 2.53791C4.40543 2.53791 4.33024 2.55288 4.2601 2.58197C4.18996 2.61106 4.12625 2.6537 4.07261 2.70745V2.70745C3.97079 2.81475 3.91404 2.95703 3.91404 3.10495C3.91404 3.25287 3.97079 3.39515 4.07261 3.50245L8.21261 8.12995L7.68761 8.66245L3.82511 4.82995C3.77306 4.77213 3.70945 4.7259 3.63838 4.69425C3.56732 4.66261 3.4904 4.64625 3.41261 4.64625C3.33482 4.64625 3.2579 4.66261 3.18683 4.69425C3.11577 4.7259 3.05215 4.77213 3.00011 4.82995V4.82995C2.90938 4.93381 2.85938 5.06704 2.85938 5.20495C2.85938 5.34286 2.90938 5.47609 3.00011 5.57995L6.60011 9.74995L6.07511 10.2824L3.00011 7.67995C2.89556 7.59068 2.76259 7.54163 2.62511 7.54163C2.48763 7.54163 2.35466 7.59068 2.25011 7.67995V7.67995C2.19636 7.73359 2.15373 7.7973 2.12463 7.86744C2.09554 7.93758 2.08057 8.01277 2.08057 8.0887C2.08057 8.16463 2.09554 8.23982 2.12463 8.30996C2.15373 8.38009 2.19636 8.44381 2.25011 8.49745L5.02511 11.2649L9.27011 15.5099C9.40942 15.6494 9.57485 15.7601 9.75695 15.8355C9.93904 15.911 10.1342 15.9499 10.3314 15.9499C10.5285 15.9499 10.7237 15.911 10.9058 15.8355C11.0879 15.7601 11.2533 15.6494 11.3926 15.5099L15.1426 11.7974L15.5476 11.3924C15.7446 11.195 15.8827 10.9465 15.9463 10.6749C16.0099 10.4033 15.9965 10.1194 15.9076 9.85495L13.5001 3.88495L13.0501 4.03495C12.9466 4.07147 12.8557 4.13672 12.7879 4.22305C12.7202 4.30939 12.6785 4.41324 12.6676 4.52245L12.9676 7.59745L10.8451 10.7849L10.3126 10.2524Z"
                  fill="#222222"
                />
              </svg>
            </span>
          </div>
          <div
            className="icon-connector grid-3"
          />
          <div
            className="icon-dates grid-3"
          >
            Feb 2
          </div>
          <span
            className="dotted-shipping grid-3"
          >
            Order placed
          </span>
          <span
            className="hide"
          >
            <span
              className="tooltip-container grid-3"
            >
              <span
                className="tooltip-content"
              >
                After you place your order,
                <span>
                   
                </span>
                
                <span>
                   
                </span>
                will take 1 business day to prepare it for shipment.
              </span>
            </span>
          </span>
        </div>
        <div
          className="shipped-grid-0-1 grid-1"
          id="shipped-grid-0-1"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="icon-connector-2 grid-1"
          />
          <div
            className="grid-1"
            id="grid-0-1-content"
          >
            <span
              className="etsy-icon grid-1"
            >
              <svg
                aria-hidden="true"
                className="grid-1"
                focusable="false"
                viewBox="-5 13 1.4 1.4"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.868,11.5l-4-7A1,1,0,0,0,17,4H5A1,1,0,0,0,4,5V6H2A1,1,0,1,0,2,8H6a1,1,0,0,1,0,2H3a1,1,0,0,0,0,2H5a1,1,0,1,1,0,2H4v3a1,1,0,0,0,1,1H6.05a2.5,2.5,0,0,0,4.9,0h4.1a2.5,2.5,0,0,0,4.9,0H21a1,1,0,0,0,1-1V12A1,1,0,0,0,21.868,11.5ZM8.5,19A1.5,1.5,0,1,1,10,17.5,1.5,1.5,0,0,1,8.5,19Zm5.488-8V6h1.725l2.845,5h-4.57ZM17.5,19A1.5,1.5,0,1,1,19,17.5,1.5,1.5,0,0,1,17.5,19Z"
                />
              </svg>
            </span>
          </div>
          <span
            className="icon-connector-3 grid-1"
          />
          <div
            className="icon-dates grid-1"
          >
            Feb 3 - 4
          </div>
          <span
            className="dotted-shipping grid-1"
          >
            Order ships
          </span>
          <span
            className="hide"
          >
            <span
              className="tooltip-container grid-1"
            >
              <span
                className="tooltip-content"
              >
                
                <span>
                   
                </span>
                puts your order in the mail.
              </span>
            </span>
          </span>
        </div>
        <div
          className="shipped-grid-0-2 grid-2"
          id="shipped-grid-0-2"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="icon-connector-4 grid-2"
          />
          <div
            className="grid-2"
            id="grid-0-2-content"
          >
            <span
              className="etsy-icon grid-2"
            >
              <svg
                aria-hidden="true"
                className="grid-2"
                focusable="false"
                viewBox="-5 15 1.4 1.4"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21,9.25A1.25,1.25,0,0,0,19.75,8H12.41l4.29-4.29a1,1,0,0,0-1.41-1.41L12,5.59,10.71,4.29A1,1,0,0,0,9.29,5.71L11.59,8H4.25A1.25,1.25,0,0,0,3,9.25V15H4v5.75A1.25,1.25,0,0,0,5.25,22h13.5A1.25,1.25,0,0,0,20,20.75V15h1ZM19,10v3H13V10ZM5,10h6v3H5ZM6,20V15h5v5Zm12,0H13V15h5Z"
                />
              </svg>
            </span>
          </div>
          <div
            className="icon-dates grid-2"
          >
            Feb 5 - 11
          </div>
          <span
            className="dotted-shipping grid-2"
          >
            Delivered
          </span>
          <span
            className="hide"
          >
            <span
              className="tooltip-container grid-2"
            >
              <span
                className="tooltip-content"
              >
                Estimated to arrive at your doorstep
                <span>
                   
                </span>
                Feb 5 - Feb 11
              </span>
            </span>
          </span>
        </div>
      </div>
      <div>
        <div>
          <div
            className="hide"
          >
            <div>
              <div
                className="cost-info-text"
              >
                Cost to ship
              </div>
              <div
                className="cost-info-price"
              >
                
              </div>
            </div>
          </div>
          <button
            className="shippingCostCollapsible"
            onClick={[Function]}
            type="button"
          >
            <span
              className="shippingcost-button-content"
            >
              Get shipping cost
            </span>
            <span
              className="button-arrow-up"
            />
          </button>
          <span />
          <span
            id="shippingInfo"
          >
            Ships from United States
          </span>
        </div>
        <div
          id="policy-modal"
        >
          <button
            id="policy-button"
            onClick={[Function]}
            type="button"
          >
            View shop policies
          </button>
          <div
            className="hide"
          >
            <div
              id="modal-close-button"
            >
              <button
                className="close-button"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="etsy-icon-button"
                >
                  <svg
                    aria-hidden="true"
                    focusable="false"
                    viewBox="11 0 1.8 1.8"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.414,12l6.293-6.293a1,1,0,0,0-1.414-1.414L12,10.586,5.707,4.293A1,1,0,0,0,4.293,5.707L10.586,12,4.293,18.293a1,1,0,1,0,1.414,1.414L12,13.414l6.293,6.293a1,1,0,0,0,1.414-1.414Z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              className="modal-container"
            >
              <div
                className="modal-content"
              >
                <span
                  className="main-title"
                >
                  Shop policies for
                  <span>
                     
                  </span>
                  
                </span>
                <br />
                <span
                  className="small-grey-text"
                >
                  Last updated on
                  <span>
                     
                  </span>
                  Invalid date
                </span>
                <br />
                <br />
                <span />
                <span
                  className="title"
                >
                  Payments
                </span>
                <div>
                  <svg
                    aria-hidden="true"
                    focusable="false"
                    viewBox="0 0 21 21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M17,10V7A5,5,0,0,0,7,7v3H5v8a2,2,0,0,0,2,2H17a2,2,0,0,0,2-2V10H17Zm-4,7a1,1,0,0,1-2,0V13a1,1,0,0,1,2,0v4Zm2-7H9V7a2.935,2.935,0,0,1,3-3,2.935,2.935,0,0,1,3,3v3Z"
                    />
                  </svg>
                  <span
                    className="payment-content"
                  >
                    Secure options
                    <img
                      alt="Screen-Shot-2021-01-21-at-3-49-08-PM"
                      border="0"
                      src="https://i.ibb.co/mzGDryy/Screen-Shot-2021-01-21-at-3-49-08-PM.png"
                    />
                    Accepts Etsy Gift Cards and Etsy Credits
                    <br />
                    Etsy keeps your payment information secure.
                    <span>
                       
                    </span>
                    Etsy shops never receive your credit card information.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
